lib/grape-swagger/doc_methods.rb

```diff
@@ -501,51 +504,57 @@ def setup(options)
 
             summary = @@documentation_class.translate(
-              route.route_description, i18n_scope,
+              route.description, i18n_scope,
               [:"#{i18n_key}.desc", :"#{i18n_key}.description"]
             )
             notes = @@documentation_class.translate(
-              route.route_detail || route.route_notes, i18n_scope,
+              route_settings_description[:detail] || route_settings_description[:notes], i18n_scope,
               [:"#{i18n_key}.detail", :"#{i18n_key}.notes"]
             )
             notes       = @@documentation_class.as_markdown(notes)
 
-            http_codes  = @@documentation_class.parse_http_codes(route.route_http_codes, models)
+            http_codes  = @@documentation_class.parse_http_codes(route.http_codes, models)
 
-            models.merge(Array(route.route_entity)) if route.route_entity.present?
+            models.merge(Array(route.entity)) if route.entity.present?
+
+            route_settings_description = route.settings[:description] || {}
 
             operation = {
               notes: notes.to_s,
               summary: summary,
-              nickname: route.route_nickname || (route.route_method + route.route_path.gsub(/[\/:\(\)\.]/, '-')),
-              method: route.route_method,
-              parameters: @@documentation_class.parse_header_params(route.route_headers, scope: i18n_scope, key: i18n_key) +
-                          @@documentation_class.parse_params(route.route_params, route.route_path, route.route_method,
+              nickname: route_settings_description[:nickname] || (route.request_method + route.path.gsub(/[\/:\(\)\.]/, '-')),
+              method: route.request_method,
+              parameters: @@documentation_class.parse_header_params(route.headers, scope: i18n_scope, key: i18n_key) +
+                          @@documentation_class.parse_params(route.route_params, route.path, route.request_method,
                                                              scope: i18n_scope, key: i18n_key),
-              type: route.route_is_array ? 'array' : 'void'
+              type: route_settings_description[:is_array] ? 'array' : 'void'
             }
-            operation[:authorizations] = route.route_authorizations unless route.route_authorizations.nil? || route.route_authorizations.empty?
+
+            authorizations = route_settings_description[:authorizations]
+            operation[:authorizations] = authorizations if authorizations && authorizations.any?
+
             if operation[:parameters].any? { |param| param[:type] == 'File' }
               operation.merge!(consumes: ['multipart/form-data'])
             end
             operation.merge!(responseMessages: http_codes) unless http_codes.empty?
 
-            if route.route_entity
-              type = @@documentation_class.parse_entity_name(Array(route.route_entity).first)
-              if route.route_is_array
+            if route.entity
```

Comment:
Actually no need, these were declared as DSL sugar on top of the `entity` and `http_codes` implementation in https://github.com/ruby-grape/grape/blob/1097eb192a8601234b5465c23b1ab09101048792/lib/grape/dsl/desc.rb#L90, so they work today out of the box.